### ISSUE TITLE
Pass additional metadata to policy evaluations

### ISSUE THREAD

**Issue by mattmoor** [sigstore/cosign#1232]
**Description**

Today the policy validation is only passed the body of a single statement for evaluation, after the signature has been verified.

This makes some sense when the attestation is being verified against a particular key, but for keyless mode part of the policy check MUST include the `iss` and `sub`, but it should also include all of the metadata we encode in the certificate extension(s).

**Comment by JimBugwadia** [comment]
For keyless, would also be good to pass claims returned from OIDC scopes as discussed here: https://github.com/sigstore/cosign/issues/1087.

**Comment by mattmoor** [comment]
Yeah, I expect that'd get stored in one of the extensions, which we should pass in as well.  I alluded to this in a TODO of the other issue I opened (linked to this)
